tests/test_doctestplus.py

```diff
@@ -865,6 +865,28 @@ def test_doctest_subpackage_requires(subpackage_requires_testdir, caplog):
     assert caplog.text == ''
 
 
+def test_import_mode(testdir):
+    """Test that two files with the same name but in different folders work with --import-mode=importlib."""
+    a = testdir.mkdir('a')
+    b = testdir.mkdir('b')
+
+    pyfile = dedent("""
+        def f():
+            '''
+            >>> 1
+            1
+            '''
+    """)
+
+    a.join('testcode.py').write(pyfile)
+    b.join('testcode.py').write(pyfile)
+
+    mismatch = testdir.inline_run(str(testdir), "--doctest-plus", "--import-mode=prepend")
```

Comment:
What about testing for `append`?

Reply:
Also this syntax appears to not work for some pytest versions. Maybe you can skip patching for pytest<6 unless you care about old pytest support?